src/server/internalServer.cpp

```diff
@@ -366,7 +362,8 @@ zim::Query SearchInfo::getZimQuery(bool verbose) const {
   if (verbose) {
     std::cout << "Performing query '" << pattern<< "'";
   }
-  query.setQuery(pattern);
+  const std::string finalPattern = kiwix::trim(pattern);
+  query.setQuery(finalPattern);
```

Comment:
An extra variable is not justified. Call `kiwix::trim()` inline in the expression.